metamorph/server.go

```diff
@@ -37,11 +36,18 @@ func init() {
 	gocore.NewStat("PutTransaction", true)
 }
 
+// To returns a pointer to the given value.
+func PtrTo[T any](v T) *T {
+	return &v
+}
+
 const (
 	responseTimeout = 5 * time.Second
 	blocktxTimeout  = 1 * time.Second
 )
 
+var ErrNotFound = errors.New("key could not be found")
```

Comment:
I think you'll have to add this, otherwise this error has no effect.

```
func (s *Server) GetTransactionStatus(ctx context.Context, req *metamorph_api.TransactionStatusRequest) (*metamorph_api.TransactionStatus, error) {
	data, announcedAt, minedAt, storedAt, err := s.getTransactionData(ctx, req)
	if err != nil {
		
		if errors.Is(err, store.ErrNotFound){
			return nil, ErrNotFound
		}
           ...
```